### DISCUSSION TITLE
Can ChatBot in Angular 10 be migrated to Angular 18?

### DISCUSSION THREAD

**Discussion by DvorakDovahkiin** [clientIO/joint#2348] [Q&A]
### Introduction

I'm new to using JointJS and I'm trying to run the ChatBot example project in Angular 18, but I got a series of deprecated object warnings and finally an error that prevents me from running the project. What would be the recommendation to be able to execute the project?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
Is _Angular 18_ a thing yet? Do you mean version _16_?

**Reply by DvorakDovahkiin** [reply]
Yeah.  Sorry.  16.

**Reply by kumilingus** [reply]
We have updated our [Angular tutorial](https://github.com/clientIO/joint-plus-tutorial-angular) to version 16.
The _ChatBot_ application will be upgraded next. Please bear with us.

**Comment by DvorakDovahkiin** [comment]
Thanks a lot.

De: Roman Bruckner ***@***.***>
Enviado el: miércoles, 27 de septiembre de 2023 4:05
Para: clientIO/joint ***@***.***>
CC: Moisés Díaz Salazar ***@***.***>; Author ***@***.***>
Asunto: Re: [clientIO/joint] Can ChatBot in Angular 10 be migrated to Angular 18? (Discussion #2348)


We have updated our Angular tutorial<https://github.com/clientIO/joint-plus-tutorial-angular> to version 16.
The ChatBot application will be upgraded next. Please bear with us.
